### PR TITLE
RedisFeatureStore: enable local cache for All()

### PR DESCRIPTION
**Pull request by drichelson**
see writeup: https://catamorphic.quip.com/dah4AU3Xkoas

### CODE REVIEW

**Comment by jkodumal** [comment]
Are you missing an invalidation on delete?

File: redis.go [hunk 1]
```diff
@@ -109,6 +116,9 @@ func (store *RedisFeatureStore) Get(key string) (*FeatureFlag, error) {
 				if feature.Deleted {
 					store.logger.Printf("RedisFeatureStore: WARN: Attempted to get deleted feature flag (from local cache). Key: %s", key)
 					return nil, nil
+				} else {
```
Comment (jkodumal) [inline redis.go]: This isn't right-- this is the else branch on if the feature is deleted, right?

File: redis.go [hunk 2]
```diff
@@ -19,7 +20,13 @@ type RedisFeatureStore struct {
 	logger  *log.Logger
 }
 
-const initKey = "$initialized$"
+const (
+	initKey = "$initialized$"
+
+	// stored alongside flag keys.
+	// # is not a valid character for flag keys, so there will never be a collision.
+	allFlagsKey = "#all_flags"
```
Comment (jkodumal) [inline redis.go]: We're already using `$special_key$` as a convention-- can we stick with that?

Comment (drichelson) [inline redis.go]: yes

File: redis.go (hunk 1 above)
Comment (drichelson) [inline redis.go]: good catch

**Comment by drichelson** [comment]
@jkodumal regarding deletes: 
For single flag changes: all write paths end up calling put() where we delete all flags from the local cache.

**Comment by jkodumal** [comment]
👍 

LGTM once the integration harnesses are bumped and tested with these changes.

**Comment by drichelson** [comment]
integration harness works locally in redis mode.

**Pull request merged**